### ISSUE TITLE
`rotate{90,180.270}` methods do not allow avoiding copies when destination buffer already exists

### ISSUE THREAD

**Issue by nagisa** [image-rs/image#1082]
Ignoring the issue template, because this is a very free-form bug.

The current `imageops::rotate{90,180,270}` functions have the following signature:

```
pub fn rotate{90,180,270}<I: GenericImageView>(
    image: &I
) -> ImageBuffer<I::Pixel, Vec<<I::Pixel as Pixel>::Subpixel>> where
    I::Pixel: 'static, 
```

However if one already has a destination buffer, then a copy becomes necessary and an unnecessary allocation occurs. I believe these functions ought to be:

```
pub fn rotate{90,180,270}<I, Container>(
    image: &I,
    destination: &mut ImageBuffer<I::Pixel, Container>
) where
    I: GenericImageView,
    I::Pixel: 'static, 
    Container: Deref<Target = [P::Subpixel]> 
```

or something similar to allow this use-case.

**Comment by HeroicKatora** [comment]
Not to forget, rotate in-place which the proposed functio could also not do.

**Comment by nagisa** [comment]
Rotate in place is a harder problem to solve in general, I think, and would require a less naive algorithm to work at all, but a fair point.

**Comment by Veykril** [comment]
This has been addressed in #1102, the only operations that are missing are 90 and 270 degree in place rotations since those were rather non-trivial(if even possible, since these change an image's dimensions).
